tests/app/test_base.py

```diff
@@ -1,262 +1,164 @@
-from unittest.mock import patch, call
-
 import pytest
+from pytest_mock import MockerFixture
 
 from corva.app.base import BaseApp
-from corva.app.base import ProcessResult
-from corva.event.base import BaseEvent
 from corva.event.data.base import BaseEventData
-from corva.utils import get_state_key
+from corva.event.event import Event
+from tests.conftest import ComparableException, APP_KEY, CACHE_URL
+
+
+@pytest.fixture(scope='function')
+def base_app(mocker: MockerFixture):
+    # patch __abstractmethods__, so we can initialize BaseApp
+    mocker.patch.object(BaseApp, '__abstractmethods__', set())
+
+    return BaseApp(app_key=APP_KEY, cache_url=CACHE_URL)
 
 
 def test_abstractmethods():
-    assert getattr(BaseApp, '__abstractmethods__') == frozenset(['event_cls'])
     with pytest.raises(TypeError):
         BaseApp()
 
+    assert (
+         getattr(BaseApp, '__abstractmethods__')
+         ==
+         frozenset([
+             'event_loader',
+             'group_by_field',
+             'get_context'
+         ])
+    )
```

Comment:
why do we need a check of which methods are abstract?

Reply:
We test them to verify, that we didn't forget to make selected methods abstract and that they must be overridden in child  classes

Reply:
We don't need to verify the structure and signature of the class or its methods. It's static.
Generally, when we write the test when we need to test:
* logic
* interface (parts of the application which will be exposed to the user or to other components of the system)
And we do not need to test:
* ourselves (how do we write code and what methods we use)
* implementation details (how a programming language works, how libraries works)

Here is an example of the possible scenario: one of the child classes doesn't really need this method to be implemented and we remove `abstractmethod` and use `return NotImplemented` instead. Logically, nothing should break. Though, this test will fail.

Reply:
Thanks, deleted the test